src/axom/core/ArrayBase.hpp

```diff
@@ -315,10 +315,13 @@ class ArrayBase
   /// \brief Set the shape
   AXOM_HOST_DEVICE void setShape(const StackArray<IndexType, DIM>& shape_)
   {
+#ifndef NDEBUG
```

Comment:
My understanding is that `NDEBUG`  (i.e. "no debug") disables asserts, so although it looks awkward, I think this is correct.

I wanted to use either `AXOM_DEBUG` or `SLIC_ASSERT`, but we can control the former independently of `C++`  asserts, and our `core` component cannot use `slic` since `slic` depends on `core`.

See: https://stackoverflow.com/questions/2290509/debug-vs-ndebug

Reply:
Got it. We should rename the macro and use #if defined(...) and #if !defined(...).